src/bundles/communication/Communications.ts

```diff
@@ -0,0 +1,123 @@
+import context from 'js-slang/context';
+import { MultiUserController } from './MultiUserController';
+import { GlobalStateController } from './GlobalStateController';
+import { RpcController } from './RpcController';
+
+class CommunicationModuleState {
+  multiUser: MultiUserController;
+  globalState: GlobalStateController | null = null;
+  rpc: RpcController | null = null;
+
+  constructor(address: string, port: number) {
+    let multiUser = new MultiUserController();
+    multiUser.setupController(address, port);
+    this.multiUser = multiUser;
+  }
+}
+
+let moduleState = new CommunicationModuleState('broker.hivemq.com', 8884);
+context.moduleContexts.communication.state = moduleState;
+
+// Loop
+
+let interval: number | undefined;
+
+/**
+ * Keeps the program running so that messages can come in.
+ */
+export function keepRunning() {
+  interval = window.setInterval(() => {}, 20000);
+}
+
+/**
+ * Removes interval that keeps the program running.
+ */
+export function stopRunning() {
+  if (interval) {
+    window.clearInterval(interval);
+    interval = undefined;
+  }
+}
+
+// Global State
+
+/**
+ * Initializes global state.
+ *
+ * @param topicHeader MQTT topic to use for global state.
+ * @param callback Callback to receive updates of global state.
+ */
+export function initGlobalState(
```

Comment:
Since this is to be exported, might be good to add a guard clause to prevent it being called a second time (perhaps throw an error to the user)?

Reply:
ok